src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

```diff
@@ -540,9 +547,16 @@ private AIState executeNodeMining()
 
     private AIState searchANodeToMine(@NotNull final Level currentLevel)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "searchANodeToMine" is 17 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)